map/framework.cpp

```diff
@@ -1463,16 +1472,13 @@ bool Framework::Search(search::SearchParams const & params)
   if (ParseEditorDebugCommand(params))
     return true;
 
-  m2::RectD const viewport = GetCurrentViewport();
-
-  if (QueryMayBeSkipped(intent, rParams, viewport))
+  if (QueryMayBeSkipped(intent, rParams, GetCurrentViewport()))
     return false;
 
   intent.m_params = rParams;
-  intent.m_viewport = viewport;
-
   // Cancels previous search request (if any) and initiates new search request.
   CancelQuery(intent.m_handle);
+  intent.m_isDelayed = false;
```

Comment:
OnViewportChanged() is called on the UI thread, Search is always called on the UI thread. So, I didn't get your question, please, paraphrase it.